pkg/kubeinteraction/resultsannotation.go

```diff
@@ -0,0 +1,48 @@
+package kubeinteraction
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params/info"
+	tektonv1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+)
+
+const (
+	resGroupName            = "results.tekton.dev"
+	resAnnotation           = "/resAnnotations"
+	recordSummaryAnnotation = "/recordSummaryAnnotations"
+)
+
+type ResultAnnotation struct {
+	Repo          string `json:"repo"`
+	Commit        string `json:"commit"`
+	EventType     string `json:"eventType"`
+	PullRequestID int    `json:"pull_request-id"`
+}
+
+// Add annotations to PipelineRuns produced by PaC for TektonResults
+// to capture data for summary and record.
+func AddResultsAnnotation(event *info.Event, pipelineRun *tektonv1.PipelineRun) error {
+	if event == nil {
+		return fmt.Errorf("nil event")
+	}
+	resultAnnotation := ResultAnnotation{
+		Repo:          event.Repository,
+		Commit:        event.SHA,
+		EventType:     event.EventType,
+		PullRequestID: event.PullRequestNumber,
```

Comment:
for push event this would be 0 I think, probably better to check if it exist and then set

Reply:
If we add omitempty to the PullRequestID's tag, then marshaling should omit that field.

Reply:
sure, that's a good idea Khurram.